game_management/game_input_handler.py

```diff
@@ -0,0 +1,560 @@
+        match game_state.board.overlay.pause_option:
+            case PauseOption.RESUME:
+                game_state.board.overlay.toggle_pause()
+            case PauseOption.SAVE:
+                save_game(game_state)
+                game_state.board.overlay.has_saved = True
+            case PauseOption.CONTROLS:
+                game_state.board.overlay.show_additional_controls = False
+                game_state.board.overlay.toggle_controls()
+            case PauseOption.QUIT:
+                game_state.game_started = False
+                game_state.on_menu = True
+                game_controller.menu.loading_game = False
+                game_controller.menu.in_game_setup = False
+                game_controller.menu.main_menu_option = MainMenuOption.NEW_GAME
+                game_controller.music_player.stop_game_music()
+                game_controller.music_player.play_menu_music()
+    elif game_state.game_started and not (
+            game_state.board.overlay.is_tutorial() or game_state.board.overlay.is_deployment() or
+            game_state.board.overlay.is_bless_notif() or
+            game_state.board.overlay.is_constr_notif() or game_state.board.overlay.is_lvl_notif() or
+            game_state.board.overlay.is_close_to_vic() or
+            game_state.board.overlay.is_investigation() or game_state.board.overlay.is_night()):
+        # If we are not in any of the above situations, end the turn.
+        if game_state.end_turn():
+            # Autosave every 10 turns.
+            if game_state.turn % 10 == 0:
+                save_game(game_state, True)
+
+            game_state.board.overlay.update_turn(game_state.turn)
+            game_state.process_heathens()
+            game_state.process_ais(game_controller.move_maker)
+
+
+def on_key_shift(game_state: GameState):
+    """
+    Handles a Shift key event in the game loop.
+    :param game_state: The current GameState object.
+    """
+    if game_state.game_started:
+        game_state.board.overlay.remove_warning_if_possible()
+        # Display the standard overlay.
+        game_state.board.overlay.toggle_standard(game_state.turn)
+
+
+def on_key_c(game_state: GameState):
+    """
+    Handles a C key event in the game loop.
+    :param game_state: The current GameState object.
+    """
+    if game_state.game_started and game_state.board.selected_settlement is not None:
+        # Pick a construction.
+        game_state.board.overlay.toggle_construction(get_available_improvements(game_state.players[0],
+                                                                                game_state.board.selected_settlement),
+                                                     PROJECTS,
+                                                     get_available_unit_plans(game_state.players[0],
+                                                                              game_state.board.selected_settlement.level))
```

Comment:
This plus a few other lines in this file are too long - it escapes the lint because the current pylint command doesn't search through sub-directories. You should be able to update the CI to `pylint $(git ls-files '*.py')` to fix that.

Reply:
Nice, I've added that to the ci now, local testing with the '' didn't work but without it, it's fine. Probably just Windows-jank though

Reply:
Interestingly, without the quotes it doesn't include subdirectories. Also weird that it didn't work with the quotes on your Windows machine; I've used that same exact command for CI on GitHub on `windows-latest` before.

Reply:
Ah whack! I'll slap the quotes on in that case. Probably something cooked on my end

Reply:
![image](https://user-images.githubusercontent.com/13777223/204517690-e160fb7c-6ca1-4439-85e0-f79088d628e6.png)

Reply:
Weird, but at least it's working now.